kroxylicious-additional-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/DekContext.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import java.nio.ByteBuffer;
+
+/**
+ * A DekContext encapsulates an encryptor
+ * @param <K> The type of the KEK id
+ */
+final class DekContext<K> {
+    private final AesGcmEncryptor encryptor;
+    private final ByteBuffer prefix;
+    private final K kekId;
+
+    DekContext(K kekId,
+               ByteBuffer prefix,
+               AesGcmEncryptor encryptor) {
+        this.kekId = kekId;
+        this.prefix = prefix;
+        this.encryptor = encryptor;
+    }
+
+    public K kekId() {
+        return kekId;
+    }
+
+    /**
+     * Returns the size of the encoding of a plaintext of the given size
+     * @param plaintextSize The plaintext.
+     * @return The size, in bytes, of a plaintext.
+     */
+    public int encodedSize(int plaintextSize) {
+        return prefix.capacity() + encryptor.outputSize(plaintextSize);
```

Comment:
we shouldn't assume the bytebuffer has been sized exactly.
```suggestion
        return prefix.remaining() + encryptor.outputSize(plaintextSize);
```